### PR TITLE
GitHub Actions test-integration Step 1: Move ant QA tools to Composer test

### PR DESCRIPTION
**Pull request by MKodde**
In this pull request all of the ant tests are now ran using composer scripts.
Therefore the ant dependencies have all been removed.
However travis ci wont build successfully due to a yarn warning, this is not a pressing issue.

### CODE REVIEW

File: ci/qa/phpmd [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+cd $(dirname $0)/../../
+
+./vendor/bin/phpmd src text ci/qa/phpmd.xml --exclude */Tests/*
```
Comment (MKodde) [inline ci/qa/phpmd:5]: Please add a linefeed to this file. As a convention all files end with an empty line. :man_shrugging:

File: ci/qa/phpmd.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    <rule ref="rulesets/controversial.xml" />
+
+    <rule ref="rulesets/design.xml" />
+
+    <rule ref="rulesets/naming.xml">
+        <exclude name="ShortVariable" />
+        <exclude name="LongVariable" />
+    </rule>
+
+</ruleset>
```
Comment (MKodde) [inline ci/qa/phpmd.xml:35]: I'd expect a move of the /phpmd.xml file. That way you preserve the version history of the file! If you'd like I can show you how.

**Comment by MKodde** [comment]
## Feedback
Great work so far. I see a lot of good stuff!

Regarding writing commit messages. Always finish this sentence: 'If applied, this commit will..'. That way the
message will be in the present tense and be imperative.

Example: '_If applied, this commit will_': Version pin symfony/symfony to version 4.4.*
Or: '_If applied, this commit will_': Create a new bash script that runs PHP MD

Of course the 'If applied...' is not part of the commit message, but you understand that. 

For seven straight forward rules you should follow: https://gist.github.com/julienbourdeau/e605e4b8b47da97c249a0f72598529c8
For a longer, excellent read: https://cbea.ms/git-commit/

## Commit messages
I cherry picked a couple of your commits and added some suggestions to improve in the future.

### Structured test-integration script  e91ae7f
- The commit message should be imperative, present tense, so check :heavy-check-mark:
- A little explanation could have added to the why this commit is beneficial. Otherwise, feel free to leave the changed
  file in your local staging until other changes make a commit more relevant. This commit feels like a work in progress
  thingy. You should aim to commit finished work.

### Added bash script 70703564
- The commit message should be imperative, present tense. Failed that requirement here
- Consider something more expressive like: `Create the phplint script to run from composer test`

### Syntax check of sourcecode 7183028e
- The commit message should be imperative, present tense. Failed that requirement here
- For example write: Check syntax of source code.

## Finally
The quality of the commit messages is not always up to par yet. However, It's better than I've seen from others before :)
One of the most important things (I think) you can improve on is to only commit work when it's done. That way the commit
history is nice and clean! Sometimes this is not practical. Then feel free to do what works best for you.

I really enjoy the progress you are making on multiple levels!

File: composer.json [hunk 3]
```diff
@@ -53,7 +53,7 @@
         "squizlabs/php_codesniffer": "^3.4",
         "symfony/dotenv": "4.4.*",
         "symfony/maker-bundle": "^1.19",
-        "symfony/phpunit-bridge": "^3.0",
+        "symfony/phpunit-bridge": "^6.0",
```
Comment (MKodde) [inline composer.json]: Can you recall why this version was updated to `^6.0`?

**Review by MKodde — COMMENTED** [review]
Almost there! 

Thanks for all the great work again! Very cool to see `ant` disapear! It's also good to see that the quality of your commit messages have been improved immensely!

Some work remains:

1. As mentioned before. Please add a line feed to all files that now do not have one. You can easily spot them in the `changed-files` section here on github. They have red sign on the last row of that file.
2. PHP CPD used to be an allowed failure. But now the build halts on that step. Please investigate to reporduce that behavior in the new setup. Or alternatively fix the code duplication issue (but that should be done on a separate PR!)

**Comment by VadimSchmitz** [comment]
resolved all the comments

File: composer.json (hunk 3 above)
Comment (MKodde) [reply composer.json]: We decided to leave this as is for now as it will not harm us

File: ci/qa/phplint.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+path: [./src, ./tests]
+jobs: 10
+cache: /var/qa/phplint.cache
+extensions:
+  - php
+exclude:
+  - vendor
+  -
```
Comment (MKodde) [inline ci/qa/phplint.yaml:8]: So close.

File: ci/qa/phpcpd [hunk 5]
```diff
@@ -4,3 +4,4 @@ cd $(dirname $0)/../../
 
 # https://github.com/sebastianbergmann/phpcpd
 ./vendor/bin/phpcpd --exclude src/*/Tests/* src
+exit 0
```
Comment (MKodde) [inline ci/qa/phpcpd:7]: Nice and to the point!

**Pull request merged**